airflow/providers/oracle/hooks/oracle.py

```diff
@@ -90,6 +134,49 @@ def get_conn(self) -> oracledb.Connection:
         mod = conn.extra_dejson.get('module')
         schema = conn.schema
 
+        # Enable oracledb thick mode if thick_mode is set to True, defaults to False
+        # Parameters take precedence over connection config extra
+        # Defaults to False (use thin mode) if not provided in params or connection config extra
+        if self.thick_mode is None:
+            self.thick_mode = conn.extra_dejson.get('thick_mode', False)
+            if not isinstance(self.thick_mode, bool):
+                raise TypeError(f'thick_mode expected bool, got {type(self.thick_mode).__name__}')
+        if self.thick_mode:
+            if self.thick_mode_lib_dir is None:
+                self.thick_mode_lib_dir = conn.extra_dejson.get('thick_mode_lib_dir')
+                if not isinstance(self.thick_mode_lib_dir, (str, type(None))):
+                    raise TypeError(
+                        f'thick_mode_lib_dir expected str or None, '
+                        f'got {type(self.thick_mode_lib_dir).__name__}'
+                    )
+            if self.thick_mode_config_dir is None:
+                self.thick_mode_config_dir = conn.extra_dejson.get('thick_mode_config_dir')
+                if not isinstance(self.thick_mode_config_dir, (str, type(None))):
+                    raise TypeError(
+                        f'thick_mode_config_dir expected str or None, '
+                        f'got {type(self.thick_mode_config_dir).__name__}'
+                    )
+            oracledb.init_oracle_client(
+                lib_dir=self.thick_mode_lib_dir, config_dir=self.thick_mode_config_dir
+            )
+
+        # Set oracledb Defaults Attributes
+        # Default to the initial values
+        # if not provided in params or connection config extra
+        # (https://python-oracledb.readthedocs.io/en/latest/api_manual/defaults.html)
+        if self.fetch_decimals is None:
+            self.fetch_decimals = conn.extra_dejson.get('fetch_decimals', False)
+            if not isinstance(self.fetch_decimals, bool):
+                raise TypeError(f'fetch_decimals expected bool, got {type(self.fetch_decimals).__name__}')
+        oracledb.defaults.fetch_decimals = self.fetch_decimals
```

Comment:
notice here that you are setting this default always, even when the user does not specify this param.  it's sorta ok because you are setting it to _false_, and that is what the default is in the library.  but, what if the library changed the default?  or, what if you made a mistake and the library's default isn't what you thought?  you can avoid this risk by only setting attributes that are explicitly provided by the user.  it's probably just a little more code, but, it means not doing stuff that you don't really need to do.

but actually, it can be done pretty simply... you know ... you need to both (1) convert string to bool and (2) coalesce params a bunch of times here.... probably best to do something like this:

```python
        def _get_bool(val):
            if isinstance(val, bool):
                return val
            if isinstance(val, str):
                val = val.lower().strip()
                if val == 'true':
                    return True
                if val == 'false':
                    return False
            return None
        def _get_first_bool(*vals):
            for val in vals:
                converted = _get_bool(val)
                if isinstance(converted, bool):
                    return converted
            return None
        fetch_decimals = _get_first_bool(self.fetch_decimals, extras.get('fetch_decimals'))
        if isinstance(fetch_decimals, bool):
            oracledb.defaults.fetch_decimals = fetch_decimals
```